$MirOS: ports/misc/mc/patches/patch-vfs_extfs_iso9660_in,v 1.2 2008/09/17 20:06:59 tg Exp $

	This does limit names to the C locale, but skips error messages.

--- vfs/extfs/iso9660.in.orig	Fri Oct 29 09:14:16 2004
+++ vfs/extfs/iso9660.in	Fri Dec 31 17:44:06 2010
@@ -9,15 +9,7 @@
 # tested to comply with isoinfo 2.0's output
 
 test_iso () {
-    CHARSET=`echo "$save_ctype" | sed -n -e 's/.*\.\(.*\)"$/\1/p' | tr '[A-Z]' '[a-z]'`
-    if test -n "$CHARSET"; then
-	isoinfo -j $CHARSET 2>&1 | grep "Unknown charset" >/dev/null && CHARSET=
-    fi
-    if test -n "$CHARSET"; then
-	JOLIET_OPT="-j $CHARSET -J"
-    else
 	JOLIET_OPT="-J"
-    fi
     ISOINFO="isoinfo -R"
     isoinfo -d -i "$1" | grep "NO Joliet" > /dev/null || ISOINFO="$ISOINFO $JOLIET_OPT"
 }
@@ -25,6 +17,8 @@ test_iso () {
 mcisofs_list () {
 # left as a reminder to implement compressed image support =)
 case "$1" in
+  *.xz) MYCAT="xz -dc";;
+  *.lzma) MYCAT="lzma -dc";;
   *.bz2) MYCAT="bzip2 -dc";;
   *.gz)  MYCAT="gzip -dc";;
   *.z)   MYCAT="gzip -dc";;
@@ -65,7 +59,6 @@ mcisofs_copyout () {
 	$ISOINFO -i "$1" -x "/$2" > "$3" 
 }
 
-save_ctype=`locale | grep LC_CTYPE`
 LC_ALL=C
 
 cmd="$1"
